### PR TITLE
feat(api): implement file deletion

### PR DESCRIPTION
**Pull request by caveman99**
Description

This adds a implementation of file delete to the Django API. Right now the code will only manipulate the database but leave the file in place.

No documentation change is needed as the API Documentation is auto-generated from the code.
Testing Notes

Playground server with new code successfully executed DELETE api call.

### CODE REVIEW

**Comment by jooola** [comment]
We are missing an error schema for our API error responses. Returning a simple HTTP status code is not enough.

File: api/libretime_api/storage/views/file.py [hunk 1]
```diff
@@ -29,3 +34,24 @@ def download(self, request, pk=None):  # pylint: disable=invalid-name
...
+
+    def destroy(self, request, *args, **kwargs):  # pylint: disable=invalid-name
+        pk = kwargs.get("pk", None)
+        pk = IntegerField().to_internal_value(data=pk)
+
+        file = get_object_or_404(File, pk=pk)
+
+        # Check if the file is scheduled to be played in the future
+        if Schedule.is_file_scheduled_in_the_future(file_id=pk) is True:
+            return HttpResponse(status=409)
```
Comment (paddatrapper) [inline api/libretime_api/storage/views/file.py:46]: Let's include a useful error message in these responses

**Comment by paddatrapper** [comment]
> We are missing an error schema for our API error responses. Returning a simple HTTP status code is not enough.

Do we want to define it here or merge this with basic human-readable errors for now and spend time defining it outside this?

**Comment by jooola** [comment]
We can implement a custom exceptions handler in the future, but I refactored the code to raise API exceptions instead of returning empty responses.

@caveman99 Let me know if you are ok with this implementation, and I'll be happy to merge this.

**Comment by caveman99** [comment]
@jooola looks good to me ... :-)

**Pull request merged**